graphql/schemabuilder/input.go

```diff
@@ -230,6 +235,26 @@ func (sb *schemaBuilder) getEnumArgParser(typ reflect.Type) (*argParser, graphql
 
 }
 
+// makeTextUnmarshalerParser returns an argParser that will read the passed in
+// value as a string and insert it into the destination type using the
+// encoding.TextUnmarshaler API.
+func (sb *schemaBuilder) makeTextUnmarshalerParser(typ reflect.Type) (*argParser, graphql.Type, error) {
+	return &argParser{
+		FromJSON: func(value interface{}, dest reflect.Value) error {
+			asString, ok := value.(string)
+			if !ok {
+				return errors.New("not a string")
+			}
+			unmarshalable, ok := dest.Addr().Interface().(encoding.TextUnmarshaler)
```

Comment:
Is this guaranteed to be addressable? Can this panic?

Reply:
Added a check that it's addressable (and a nicer error if it is not).